File: src/main/java/org/mozilla/msrp/platform/profile/FirefoxAccountClient.java
package org.mozilla.msrp.platform.profile;

import retrofit2.Call;
import retrofit2.http.*;

/**
 * FirefoxAccountClient talks to Firefox Account endpoints
 *
 * */
public interface FirefoxAccountClient {
    @POST("v1/token")
    Call<FxaTokenResponse> token(@Body FxaTokenRequest request);

    @GET("v1/profile")
    Call<FxaProfileResponse> profile(@Header("Authorization") String bearer);
}

File: src/main/java/org/mozilla/msrp/platform/user/FirefoxAccountClient.kt
package org.mozilla.msrp.platform.user

import retrofit2.Call
import retrofit2.http.*

/**
 * FirefoxAccountClient talks to Firefox Account endpoints
 *
 */
interface FirefoxAccountClient {
    @POST("v1/token")
    fun token(@Body request: FxaTokenRequest): Call<FxaTokenResponse>

    @GET("v1/profile")
    fun profile(@Header("Authorization") bearer: String): Call<FxaProfileResponse>
}